pyccl/halos/profiles.py

```diff
@@ -157,7 +160,9 @@ def real(self, cosmo, r, M, a, mass_def=None):
         Returns:
             float or array_like: halo profile. The shape of the
             output will be `(N_M, N_r)` where `N_r` and `N_m` are
-            the sizes of `r` and `M` respectively.
+            the sizes of `r` and `M` respectively. If `r` or `M`
+            are scalars, the corresponding dimension will be
+            squeezed out on output.
```

Comment:
What about shape `(1,)` arrays? I think those get squeezed too.

Reply:
In that case e.g. `np.ndim(M)` would be 1, and that dimension wouldn't get squeezed (hence the check for the ndims.
I'm actually testing for that [here](https://github.com/LSSTDESC/CCL/blob/330e869f5d09a79af9d687c4ba4e42f6b276712d/pyccl/tests/test_profiles.py#L24)

Reply:
Ahhhhhh ok. SGTM!